### PR TITLE
Set NODE_BINARY_TAR_URL and download them on the windows nodes

### PR DESCRIPTION
**Pull request by yujuhong**

### CODE REVIEW

File: cluster/gce/win1803/k8s-node-setup.psm1 [hunk 1]
```diff
@@ -261,15 +261,17 @@ function DownloadAndInstall-KubernetesBinaries {
...
-  Invoke-WebRequest `
-    https://storage.googleapis.com/kubernetes-release/release/${k8sVersion}/bin/windows/amd64/kubectl.exe `
-    -OutFile ${env:NODE_DIR}\kubectl.exe
-  Invoke-WebRequest `
-    https://storage.googleapis.com/kubernetes-release/release/${k8sVersion}/bin/windows/amd64/kubelet.exe `
-    -OutFile ${env:NODE_DIR}\kubelet.exe
-  Invoke-WebRequest `
-    https://storage.googleapis.com/kubernetes-release/release/${k8sVersion}/bin/windows/amd64/kube-proxy.exe `
-    -OutFile ${env:NODE_DIR}\kube-proxy.exe
+
```
Comment (yujuhong) [inline cluster/gce/win1803/k8s-node-setup.psm1]: This is some-what hacky, but I've tested and it worked.
We need to clean up more code around the directoris.
@pjh could you take a look?

Comment (pjh) [reply cluster/gce/win1803/k8s-node-setup.psm1]: I made some comments about cleaning up the directories, as you noted, but yeah this approach seems fine to me.

File: cluster/gce/win1803/k8s-node-setup.psm1 [hunk 2]
```diff
@@ -178,7 +178,7 @@ function Set-CurrentShellEnvironmentVar {
 function Set-EnvironmentVars {
   $envVars = @{
     "K8S_DIR" = "${k8sDir}"
-    "NODE_DIR" = "${k8sDir}\node"
+    "NODE_DIR" = "${k8sDir}\node\bin"
     "Path" = ${env:Path} + ";${k8sDir}\node"
```
[the remark this answers is not in the document]
Comment (yujuhong) [reply cluster/gce/win1803/k8s-node-setup.psm1]: I checked and there was no other place that needed to be changed.

File: cluster/gce/win1803/k8s-node-setup.psm1 [hunk 3]
```diff
@@ -261,15 +261,17 @@ function DownloadAndInstall-KubernetesBinaries {
...
+
+  # TODO: Verify hash
+  $uri = ${kubeEnv}['NODE_BINARY_TAR_URL']
+  # TODO: Create a shared temp directory for all downloads
+  $k8sTmpDir = "C:/tmp"
+  mkdir -Force $k8sTmpDir
+  $filename = Split-Path -leaf $uri
+  Invoke-WebRequest $uri -OutFile ${k8sTmpDir}/${filename}
+  # TODO: this needs to be the parent directory of ${env:K8S_DIR}. Derive it.
+  tar xzf ${k8sTmpDir}/${filename} -C /etc
```
[the remark this answers is not in the document]
Comment (yujuhong) [reply cluster/gce/win1803/k8s-node-setup.psm1]: Yes, that's what the TODO is for.
Ah...I wanted to untar to `/etc/kubernetes` (`K8S_DIR`), but `tar -C` needs to take the parent directory of `/etc/kubernetes`. That's why this is hard-coded as `/etc`.

**Comment by yujuhong** [comment]
/cc @pjh @mtaufen  Updated the PR. Please take a look (again). 
I'll be testing the change, so don't merge it just yet.

**Comment by pjh** [comment]
LGTM

**Comment by yujuhong** [comment]
Fix some bugs caused by rebase in the first commit.
Add a second commit to override the node_binary_tar_url only when `USE_RELEASE_NODE_BINARIES` is set. 
If this is not set, one can build windows binary locally AND use it to create a test cluster.

**Comment by yujuhong** [comment]
BTW, @pjh, this will need my PR upstream to work, so it'll wait until you sync again :-)

**Comment by pjh** [comment]
Just synced: https://github.com/pjh/kubernetes/pull/45

**Comment by yujuhong** [comment]
> Just synced: #45

Thanks!

Rebased and merging

**Pull request merged**